src/components/Button.tsx

```diff
@@ -1,6 +1,11 @@
-import type { ButtonHTMLAttributes, ReactNode } from 'react';
+'use client';
+import {
+  useTransition,
+  type ButtonHTMLAttributes,
+  type ReactNode,
+} from 'react';
 
-type ButtonVariant = 'primary' | 'secondary';
+type ButtonVariant = 'primary' | 'secondary' | 'disabled';
```

Comment:
I don't think disabled should be a variant as primary and secondary buttons can be disabled but still preserve they styling patterns. For example if primary button has background color, disabled primary button has a (greyish) background color. And if secondary button has border but not background color, disabled secondary button has (greyish) border color. Of course this depends from the design.

Reply:
https://github.com/tperamaki/yala/issues/43